src/abstractdataframe/join.jl

```diff
@@ -248,8 +248,8 @@ end
 
 function crossjoin(df1::AbstractDataFrame, df2::AbstractDataFrame)
     r1, r2 = size(df1, 1), size(df2, 1)
-    cols = [[rep(c, 1, r2) for c in columns(df1)];
-            [rep(c, r1, 1) for c in columns(df2)]]
+    cols = [[c[repeat(collect(1:length(c)), inner=[r2])] for c in columns(df1)];
```

Comment:
The new code isn't as clean as the old one here since `repeat` only works for `Array` until https://github.com/JuliaLang/julia/pull/14082 is merged. Not sure we could backport it to 0.4 or have a version in Compat.jl. It's not terrible anyway.